cas-server-core-util/src/main/java/org/jasig/cas/util/RegexUtils.java

```diff
@@ -0,0 +1,55 @@
+package org.jasig.cas.util;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+
+/**
+ * Utility class to assist with resource operations.
+ *
+ * @author Misagh Moayyed
+ * @since 4.3.0
+ */
+public final class RegexUtils {
+    private static final Logger LOGGER = LoggerFactory.getLogger(RegexUtils.class);
+
+    private RegexUtils() {}
+
+    /**
+     * Check to see if the specified pattern is a valid regular expression.
+     *
+     * @param pattern the pattern
+     * @return whether this is a valid regex or not
+     */
+    public static boolean isValidRegex(final String pattern) {
+        try {
+            LOGGER.debug("Pattern {} is a valid regex.", Pattern.compile(pattern).pattern());
+            return true;
+        } catch (final PatternSyntaxException exception) {
+            return false;
+        }
+    }
+
+    /**
+     * Concatenate all elements in the given collection to form a regex pattern.
+     *
+     * @param requiredValues  the required values
+     * @param caseInsensitive the case insensitive
+     * @return the pattern
+     */
+    public static Pattern concatenate(final Collection<String> requiredValues, final boolean caseInsensitive) {
+        final StringBuilder builder = new StringBuilder(requiredValues.size());
+        for (final String requiredValue : requiredValues) {
+            builder.append('(').append(requiredValue).append(")|");
```

Comment:
Escaping is required, not by us. By the deployer. Either the pattern is a valid regex or it's not. The deployer has to make sure it's valid. If it isn't, we don't treat it as such and we don't try to guess the deployer's intention.